packages/grid/x-data-grid/src/hooks/features/filter/useGridFilter.tsx

```diff
@@ -390,6 +390,8 @@ export const useGridFilter = (
         const tree = gridRowTreeSelector(apiRef);
         const rowIds = (tree[GRID_ROOT_GROUP_ID] as GridGroupNode).children;
         const filteredRowsLookup: Record<GridRowId, boolean> = {};
+        const cache = {};
```

Comment:
Wouldn't it be better to type it where it's defined?
```suggestion
        const cache: FilterCache = {};
```

Reply:
I don't like adding types if they're not necessary, the type system can check for correctness using inference. I've renamed the variable to `filterCache` so it's clear what its purpose is.